packages/core/src/routes/admin-user.ts

```diff
@@ -150,4 +151,20 @@ export default function adminUserRoutes<T extends AuthedRouter>(router: T) {
       return next();
     }
   );
+
+  router.delete(
+    '/users/:userId/custom-data',
+    koaGuard({
+      params: object({ userId: string().min(1) }),
+    }),
+    async (ctx, next) => {
+      const {
+        params: { userId },
+      } = ctx.guard;
+      await clearUserCustomDataById(userId);
```

Comment:
Do we need to make sure the `userId` exists first?

Reply:
We check in the `clearUserCustomDataById` function.

Reply:
First update, and then chekc the existancy, I think that can be vulnerable.

Besides, our DeletionError's status code is 500, that is a backup check.

Reply:
Good, updated.